tests/integrationv2/test_record_padding.py

```diff
@@ -0,0 +1,151 @@
+
+    for client_results in openssl.get_results():
+        client_results.assert_success()
+        # verify that the openssl is sending padded payloads with the expected size
+        assert to_bytes(expected_record_header) in client_results.stdout
+
+    expected_version = get_expected_s2n_version(protocol, provider)
+
+    for server_results in s2nd.get_results():
+        server_results.assert_success()
+        # verify that the payload was correctly received by the server
+        assert random_bytes in server_results.stdout
+        # verify that the version was correctly negotiated
+        assert to_bytes("Actual protocol version: {}".format(
+            expected_version)) in server_results.stdout
+        # verify that the cipher was correctly negotiated
+        assert to_bytes("Cipher negotiated: {}".format(
+            cipher.name)) in server_results.stdout
+
+
+@pytest.mark.uncollect_if(func=invalid_test_parameters)
+@pytest.mark.parametrize("cipher", TLS13_CIPHERS, ids=get_parameter_name)
+@pytest.mark.parametrize("provider", [OpenSSL])
+@pytest.mark.parametrize("curve", ALL_TEST_CURVES, ids=get_parameter_name)
+# only tls 1.3 supports record padding
+@pytest.mark.parametrize("protocol", [Protocols.TLS13], ids=get_parameter_name)
+@pytest.mark.parametrize("certificate", MINIMAL_TEST_CERTS, ids=get_parameter_name)
+@pytest.mark.parametrize("padding_size", PADDING_SIZES, ids=get_parameter_name)
+def test_s2n_13_client_handles_padded_records(managed_process, cipher, provider, curve, protocol, certificate,
+                                              padding_size):
+    port = next(available_ports)
+
+    random_bytes = data_bytes(PAYLOAD_SIZE)
+    server_options = ProviderOptions(
+        mode=Provider.ServerMode,
+        port=port,
+        cipher=cipher,
+        curve=curve,
+        cert=certificate.cert,
+        key=certificate.key,
+        insecure=True,
+        protocol=protocol,
+        extra_flags=[
+            '-record_padding', padding_size]
+    )
+
+    client_options = copy.copy(server_options)
+    client_options.mode = Provider.ClientMode
+    client_options.extra_flags = None
+    client_options.data_to_send = random_bytes
+
+    s2nc = managed_process(S2N, client_options, timeout=5)
+    openssl = managed_process(provider, server_options, timeout=5)
+
+    expected_version = get_expected_s2n_version(protocol, provider)
+    for client_results in s2nc.get_results():
+        client_results.assert_success()
+        assert to_bytes("Actual protocol version: {}".format(
+            expected_version)) in client_results.stdout
+        assert to_bytes("Cipher negotiated: {}".format(
+            cipher.name)) in client_results.stdout
+
+    expected_total_length = None
+    if padding_size == 0:
+        # if there is no padding, then the openssl server payload size must be the original payload size
+        # + 16 bytes of aead tag + 1 byte of content type + 1 byte for the new-line char sent by s2n
```

Comment:
I'm not following what you mean by a newline char sent from s2n. The expected total length should be in Openssl's output, so I don't understand why a byte from s2n would be in it.